azure-iot-device/tests/common/pipeline/test_pipeline_stages_mqtt.py

```diff
@@ -107,28 +107,91 @@ def transport(mocker):
 
 
 @pytest.fixture
-def op_set_connection_args(callback):
+def op_set_connection_args(mocker):
     return pipeline_ops_mqtt.SetMQTTConnectionArgsOperation(
         client_id=fake_client_id,
         hostname=fake_hostname,
         username=fake_username,
         ca_cert=fake_ca_cert,
         client_cert=fake_certificate,
-        callback=callback,
+        sas_token=fake_sas_token,
+        callback=mocker.MagicMock(),
     )
 
 
+@pytest.fixture
+def op_connect(mocker):
+    return pipeline_ops_base.ConnectOperation(callback=mocker.MagicMock())
+
+
+@pytest.fixture
+def op_reconnect(mocker):
+    return pipeline_ops_base.ReconnectOperation(callback=mocker.MagicMock())
+
+
+@pytest.fixture
+def op_disconnect(mocker):
+    return pipeline_ops_base.DisconnectOperation(callback=mocker.MagicMock())
+
+
+@pytest.fixture
+def op_publish(mocker):
+    return pipeline_ops_mqtt.MQTTPublishOperation(
+        topic=fake_topic, payload=fake_payload, callback=mocker.MagicMock()
+    )
+
+
+@pytest.fixture
+def op_subscribe(mocker):
+    return pipeline_ops_mqtt.MQTTSubscribeOperation(topic=fake_topic, callback=mocker.MagicMock())
+
+
+@pytest.fixture
+def op_unsubscribe(mocker):
+    return pipeline_ops_mqtt.MQTTUnsubscribeOperation(topic=fake_topic, callback=mocker.MagicMock())
+
+
+@pytest.fixture
+def create_transport(stage, transport, op_set_connection_args):
+    stage.run_op(op_set_connection_args)
+
+
+# TODO: This should be a package level class inherited by all .run_op() tests in all stages
+class RunOpTests(object):
```

Comment:
By inheriting these common tests around error handling, we can ensure all Run Op tests have them.

Unfortunately, this does mean that these tests end up running first within a test class, which makes reading the requirements a bit odd (they should go at the end of the other tests IMO). I'm sure there's a way this could be altered to run in the correct order via hooks, but I didn't want to waste any additional time on it right now.